src/game/features/self/AntiMelee.cpp

```diff
@@ -0,0 +1,31 @@
+#include "core/commands/LoopedCommand.hpp"
+#include "game/features/Features.hpp"
+#include "game/rdr/Natives.hpp"
+
+
+namespace YimMenu::Features
+{
+
+	class AntiMelee : public LoopedCommand
+	{
+		using LoopedCommand::LoopedCommand;
+
+		virtual void OnTick() override
+		{
+			for (int i = 0; i < 35; i++)
```

Comment:
35 different melee flags to set

Reply:
Do we have an enum somewhere?

Reply:
Nope

Reply:
Wouldn't it be wiser to set melee proof bit on the player flags instead? Using `SET_ENTITY_PROOFS` and referring to [This research](https://github.com/femga/rdr3_discoveries/tree/master/AI/ENTITY_PROOFS)

Reply:
Okay, I will fix this too.

Reply:
> Wouldn't it be wiser to set melee proof bit on the player flags instead? Using `SET_ENTITY_PROOFS` and referring to [This research](https://github.com/femga/rdr3_discoveries/tree/master/AI/ENTITY_PROOFS). Granted Its a guess since im not very experienced with Rdr and its mechanics.

Setting the proofs does not work as anticipated. the current method of setting the 35 flags is the best method. @Deadlineem can corroborate(he was able to punch me with the proof set and he was unable with the 35 flags).